javascript/javascript1/week4/script.js

```diff
@@ -0,0 +1,98 @@
+let userName = null; 
+let todoList = []; 
+
+function getReply(command) {
+  const lowerCaseCommand = command.toLowerCase();
+
+  if (lowerCaseCommand.startsWith('hello my name is ')) {
+    const name = command.slice(17); 
+    if (userName === name) {
+      return `You already introduced yourself as ${name}.`;
+    } else {
+      userName = name;
+      return `Nice to meet you ${name}`;
+    }
+  }
+
+  if (lowerCaseCommand === 'what is my name?') {
+    return userName
+      ? `Your name is ${userName}`
+      : "I don't know your name yet. Please tell me your name first.";
+  }
+
+  if (
+    lowerCaseCommand.startsWith('add ') &&
+    lowerCaseCommand.includes(' to my todo')
+  ) {
+    const todo = command.slice(4, command.indexOf(' to my todo'));
+    todoList.push(todo);
+    return `${todo} added to your todo.`;
+  }
+
+  if (
+    lowerCaseCommand.startsWith('remove ') &&
+    lowerCaseCommand.includes(' from my todo')
+  ) {
+    const todo = command.slice(7, command.indexOf(' from my todo'));
+    const index = todoList.indexOf(todo);
+    if (index > -1) {
+      todoList.splice(index, 1); 
+      return `Removed ${todo} from your todo.`;
+    } else {
+      return `${todo} is not in your todo list.`;
+    }
+  }
+
+  if (lowerCaseCommand === 'what is on my todo?') {
+    if (todoList.length === 0) {
+      return 'Your todo list is empty.';
+    } else {
+      return `You have ${todoList.length} todos: ${todoList.join(', ')}.`;
+    }
+  }
+
+  if (lowerCaseCommand === 'what day is it today?') {
+    const today = new Date();
+    const day = today.getDate();
+    const month = today.toLocaleString('default', { month: 'long' }); 
```

Comment:
Great use of some newer Javascript functionality - `toLocaleString()`. Consider using it for the whole date format though, not just the month!